apps/main/src/llamalend/queries/user-health.query.ts

```diff
@@ -0,0 +1,44 @@
+import { getLlamaMarket } from '@/llamalend/llama.utils'
+import { queryFactory, rootKeys, type UserMarketParams, type UserMarketQuery } from '@ui-kit/lib/model'
+import { userMarketValidationSuite } from '@ui-kit/lib/model/query/user-market-validation'
+import type { Decimal } from '@ui-kit/utils'
+
+// TODO: This query if for the current user health, not the change in health when repaying debt.
+// Use `market.repayHealth` for that.
+export const { useQuery: useUserHealth, invalidate: invalidateUserHealth } = queryFactory({
```

Comment:
we should have 2 queries for this imo, like done in #1670.
The final calculation could be a 3rd hook.
Alternatively, if you want to keep them together, at least call `userHealth` with `Promise.all`